collect_app/src/main/java/org/odk/collect/android/http/CollectServerClient.java

```diff
@@ -42,21 +44,20 @@
 
     private static final String HTTP_CONTENT_TYPE_TEXT_XML = "text/xml";
 
-    private static HttpInterface httpConnection;
+    private static CollectServerClient instance;
 
-    private CollectServerClient() {}
```

Comment:
Hey @xsteelej I was just looking at this class and wondering why it wasn't being injected in the locations it's needed but then I realized that the `httpConnection` is the main class that has to be mocked and injected for testing so just doing the normal singleton approach here works. The only thing that injection would have done is to make this alot more cleaner where the DI would handle the singleton creation and also if there's a time we wanted to mock the `CollectServerClient` in all locations across the app say with `Mockito` that could be done as well. Something we can discuss further and get some more eyes on.